syntax = "proto3";
import "google/protobuf/descriptor.proto";
import "google/protobuf/any.proto";
import "xmlOptions.proto";

package ops.sap;

enum ReturnInt{
    option (xml_enum_options) = {name: "RESULT_INT"};
    RETURN_INT_OK = 0 [(xml_enum_value_options)={description: "Success"}];
    RETURN_INT_UNKNOWN = 99;
}
enum StatusInt{
    option (xml_enum_options) = {name: "RESULT_INT"};
    STATUS_INT_OK = 0 [(xml_enum_value_options)={description: "Success"}];
    STATUS_INT_UNKNOWN = 99;
}
enum ProcessOk{
    OK = 0;

}

message DataType{
    string type = 1;
}
message Result
{
    message ResultData{
        message Value{
            string name = 1;
            string value = 2;
        }
        repeated Value values = 1;
    }
    optional ResultData results = 2;
    optional DataType data_type = 3;
    optional google.protobuf.Any data = 4;

    optional bool process_ok = 5;
    optional string error_text = 6;
    optional bool start_return = 7;
    oneof is_status{
        ReturnInt return_int = 8[(xml_field_options)={name:"RESULT_INT"}];
        ReturnInt status_int = 9;
    }
    oneof is_status_text{
        string return_text = 10;
        string status_text = 11;
    }
}

message Wrapper{
    google.protobuf.Any content = 1;
}

message AllInOne{
    option (xml_message_options) = {name: "TEST"};
    float flt = 1 [(xml_field_options)={name: "FLOAT"}];
    double dbl = 2;
    int32  i32 = 3;
    int64  i64 = 4;
    uint32 ui32 = 5;
    uint64 ui64 = 6;
    sint32 si32 = 7;
    sint64 si64 = 8;
    fixed32 fix32 = 9;
    fixed64 fix64 = 10;
    sfixed32 sfix32 = 11;
    sfixed64 sfix64 = 12;
    bool bl = 13;
    string str = 14;
    bytes  bts = 15;
    optional string opt = 16;
    repeated string rpt = 17;

    enum Enm{
        option (xml_enum_options) = {name: "RESULT_INT"};
        OK = 0 [(xml_enum_value_options) = {description: "Success"}];
        PARAM_ERROR = 1  [(xml_enum_value_options) = {description: "Parameter Fehler"}];
        CAM_NO_ANSWER = 2 [(xml_enum_value_options) = {description: "Kalibrier-Kamerasoftware antwortet nicht"}];
        LAMP_NO_ANSWER = 3 [(xml_enum_value_options) = {description: "Leuchte antwortet nicht"}];
        LAMP_TYPE_ERROR = 4 [(xml_enum_value_options) = {description: "Falscher Leuchtentyp Ã¼bergeben"}];
        UNKNOWN_ERROR = 99 [(xml_enum_value_options) = {description: "Unbekannter Fehler"}];
    }
    Enm enm = 18 [(xml_field_options)={name: "RESULT_INT"}];

    message ResultData{
        option (xml_message_options) ={name: "RESULT-DATA"};
        message Value{
            option (xml_message_options) ={name: "MESSWERT"};
            string name = 1 [(xml_field_options) = {name: "NAME"}];
            string value = 2 [(xml_field_options) = {name: "WERT"}];
        }
        repeated Value values = 1 [(xml_field_options) ={name: "MESSWERT"}];
    }
    ResultData result_data = 19 [(xml_field_options)={name: "RESULT-DATA"}];
}
